src/node_crypto.cc

```diff
@@ -672,8 +671,59 @@ void SecureContext::SetCert(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L && !defined(OPENSSL_IS_BORINGSSL)
+// This section contains OpenSSL 1.1.0 functions reimplemented for OpenSSL
+// 1.0.2 so that the following code can be written without lots of #if lines.
+
+static int X509_STORE_up_ref(X509_STORE* store) {
+#if OPENSSL_VERSION_NUMBER >= 0x10002000L
+  CRYPTO_add(&store->references, 1, CRYPTO_LOCK_X509_STORE);
+#else
+  // Prior to OpenSSL 1.0.2, the reference count for an X509_STORE was ignored
```

Comment:
Ok, I'll remove this. (I only included it because 1.0.1 will still be able to compile this code, but it'll corrupt memory. Hopefully the rest of this code won't compile with 1.0.1.)